internal/updater/html/parsing.go

```diff
@@ -0,0 +1,98 @@
+package htmlutils
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
```

Comment:
nit adding more error context, might be useful one day!

```suggestion
	return "", fmt.Errorf("%w: for key %q in node %s", ErrAttrNotFound, key, n)
```